smartlearner/utils.py

```diff
@@ -23,3 +25,44 @@ def save_dict_to_json_file(path, dictionary):
 def load_dict_from_json_file(path):
     with open(path, "r") as json_file:
         return json.loads(json_file.read())
+
+
+def split_dataset(dataset, proportions):
+    from .interfaces.dataset import Dataset
+    indices = np.cumsum(np.ceil(np.array(proportions) / np.sum(proportions) * len(dataset)))
+    indices[-1] = len(dataset)
+    dsets = []
+
+    for (s, f) in zip([0] + list(indices), indices):
+        s, f = int(s), int(f)
+
+        covars = dataset.inputs.get_value()[s:f]
+        targets = dataset.targets.get_value()[s:f] if dataset.targets is not None else None
+
+        dset = Dataset(covars, targets, dataset.name + '_' + str(s) + 'to' + str(f))
+        dset.symb_inputs = dataset.symb_inputs
+        dset.symb_targets = dataset.symb_targets
+
+        dsets.append(dset)
+
+    return dsets
+
+
+def kfold(dataset, k):
+    return split_dataset(dataset, np.ones((k,)))
+
+
+def sequential_kfold(dataset, k):
```

Comment:
Can you add a description of what it does, I'm not sure to understand from just looking at the code.